### ISSUE TITLE
Arena enemy pet damage not captured

### ISSUE THREAD

**Issue by crudos** [Tercioo/Details-Damage-Meter#486]
### **Before Start!**
**Make sure the bug also happen when Details! is the only addon enabled.**

**Game and Details! version**
WD 10736 RELEASE 10.0.7

**Describe the bug**
- Steps to Reproduce (what did you do to make the bug happen):
Play an arena match against a pet heavy class (BM hunter/Demo lock)

- Result (what happens when you follow the steps above):
A large portion of their damage is missing from Details. Most obvious is BM hunter where Kill Command isn't being captured by the addon.

- Expected (what you think the expected behavior would be when following the steps):
I expect to see Kill Command for the enemy hunter in Details. There are potentially other abilities being filtered but I don't know the class enough.

- When the bug started (a date, wow path release, details update, etc)?
I believe this has been the case since DF has released

- Additional information:

**Lua Error**
Post here lua errors, if you aren't getting any, use */console scriptErrors 1*

**Screenshots**
Post images of the bug if necessary.

**Combat Log**
Include a link to a combatlog file if the error is about not detecting a skill, usable item or item proc.
The log can be as simple as hitting a mob with white damage and using the item/skill.

**Other Information**
If necessary, add more info here.

**Comment by crudos** [comment]
Don't have screenshots right now but I'll add them once I get some recent examples.

**Comment by Flamanis** [comment]
Determining pet ownership and attributing it to the player is an adventure. Blizzard has no process to query ownership, so we use other methods if possible. One method we tend to use that is both reliable and slightly unreliable is using tooltips. Those generally give a GUID for the pet owner.

If you're able to, could you get a screenshot of the pet tooltip might fight, and also if the issue happens can you see if the pet attacked first? It could be due to that as well, I'm not sure.

**Comment by crudos** [comment]
Not sure if the pet attacks first but the hunter stunned (Intimidation) someone first. As soon as the pets went to me and the hunter/pets did damage he appeared on Details!. But it appeared to happen at the same time besides the initial stun.

Also note that this is only for enemy hunters. When a hunter is on my side the damage seems to be shown correctly.

Tooltips for pets mid game:
![image](https://user-images.githubusercontent.com/9119861/230703306-9731a7a3-ec22-4ff3-839e-ce77117fce70.png)
![image](https://user-images.githubusercontent.com/9119861/230703403-9ceb9580-c189-43da-8b88-9554758426e7.png)
![image](https://user-images.githubusercontent.com/9119861/230702988-20bce3da-b1f3-4629-8531-d0b3d453a36f.png)

Damage taken for my character (Crudos):
![image](https://user-images.githubusercontent.com/9119861/230703494-eb7bfd80-adc5-4a18-adc7-6bb7e29debd2.png)

Damage done by the hunter:
![image](https://user-images.githubusercontent.com/9119861/230703501-74301673-193a-434d-b2cc-42d3ca2f35d4.png)

**Comment by Flamanis** [comment]
That would likely be the cause. Could you get a `/combatlog` of an arena where this occurs?

**Comment by crudos** [comment]
I think these should be the right logs:

[combat_log_nokhudon.txt](https://github.com/Tercioo/Details-Damage-Meter/files/11182593/combat_log_nokhudon.txt)

**Comment by Flamanis** [comment]
If you are able, could you test with the changes in #496

Can provide the files directly if you wish

**Comment by crudos** [comment]
I think it's capturing demo lock but not BM hunter right now.

The "Dinosaur" pet wasn't captured under the hunter this game. Doesn't seem to be associated with the hunter but Demo lock pets are.

Damage Done:
![image](https://user-images.githubusercontent.com/9119861/231367201-01c0acf5-e4de-4adf-8eb3-52b3cdfe7138.png)

Damage Taken:
![image](https://user-images.githubusercontent.com/9119861/231367243-99c9ee38-b359-4144-af92-c7c08b0b0359.png)

**Comment by Flamanis** [comment]
And just to confirm you manually installed with the updated container_actors.lua file?

**Comment by crudos** [comment]
Yeah I just updated that 1 file that was changed and it seemed to update Demo lock damage

**Comment by Flamanis** [comment]
Alrighty. I'm gonna add a bit more 'error checking' into a new version just for testing purposes. Stuff that would let me/you see the data after the fact. I'll post a zip here sometime soon with that version.

**Comment by Flamanis** [comment]
[Details-PetFixTest2.zip](https://github.com/Tercioo/Details-Damage-Meter/files/11226111/Details-PetFixTest2.zip)

Here's a version that should give a lot of information. And by a lot of information I mean it will record how every single pet goes through the detection.

**Comment by crudos** [comment]
Sorry I've been out for a couple weeks on vacation, I'll install this one and check out what happens next time I do arena.

**Comment by Flamanis** [comment]
At this point i would not recommend. A major patch has since released and
the code has been implemented proper. Please test with most recent release

**Comment by crudos** [comment]
Will do, thanks for the quick reply.

**Comment by crudos** [comment]
It still doesn't appear to be tracking the pet damage fully. I tried `/details dumppets` but it doesn't seem to do anything.

**Comment by Flamanis** [comment]
That is because only that specific testing version had that command. What
version do you currently have installed? `/details version`

**Comment by crudos** [comment]
![image](https://github.com/Tercioo/Details-Damage-Meter/assets/9119861/0f197c06-1339-4307-ae7c-ac9c0ce6cefc)
